pkg/app/app_template_test.go

```diff
@@ -309,20 +309,21 @@ releases:
 }
 
 func TestTemplate_StrictParsing(t *testing.T) {
-	v := yaml.GoccyGoYaml
-	yaml.GoccyGoYaml = true
-	t.Cleanup(func() {
-		yaml.GoccyGoYaml = v
-	})
-
 	type testcase struct {
-		ns    string
-		error string
+		goccyGoYaml bool
+		ns          string
+		error       string
 	}
 
 	check := func(t *testing.T, tc testcase) {
 		t.Helper()
 
+		v := runtime.GoccyGoYaml
+		runtime.GoccyGoYaml = tc.goccyGoYaml
+		t.Cleanup(func() {
+			runtime.GoccyGoYaml = v
+		})
```

Comment:
This is an idiom I used widely across many test sources to ensure we modify the GoccyGoYaml runtime variable only while running the (sub-)test.